packages/core/src/shared/telemetry/vscodeTelemetry.json

```diff
@@ -527,20 +538,19 @@
             ]
         },
         {
-            "name": "stepfunctions_saveFile",
-            "description": "Called after the user saves local ASL file (inlcuding autosave) from VSCode editor or Workflow Studio",
+            "name": "stepfunctions_syncFile",
+            "description": "Triggered when Workflow Studio auto-syncs to the local file or when unsaved local changes are saved from Workflow Studio or VSCode.",
             "metadata": [
                 {
                     "type": "id",
                     "required": true
                 },
                 {
-                    "type": "saveType",
+                    "type": "stepFunctionsSyncType",
```

Comment:
why are you renaming this field? fields should not be over-specific like this

Reply:
`saveType` is a field name introduced by threat composer, which [only allows 2 values](https://github.com/aws/aws-toolkit-common/blob/0bec1b3f7af36bdbb24f38757d7c42b5dcb6928d/telemetry/definitions/commonDefinitions.json#L1687-L1694): `"MANUAL_SAVE" | "AUTO_SAVE"`.

For our use case, we want to add `auto-sync` option, as well track where the action originated (either in WFS or VSCode). If we have the latter as a separate metric, it will not be clear which specific action originated where (since we have both save and sync actions).

Do you belive there is a better way to achieve that?

Reply:
> `saveType` is a field name introduced by threat composer, which [only allows 2 values](https://github.com/aws/aws-toolkit-common/blob/0bec1b3f7af36bdbb24f38757d7c42b5dcb6928d/telemetry/definitions/commonDefinitions.json#L1687-L1694): `"MANUAL_SAVE" | "AUTO_SAVE"`.

add more values. enums are just lists that guide string values.

Reply:
I can add an action such as `AUTO_SYNC` in that list, I just wanted to make sure that can track the metric with enough granularity in that case. We have 3 scenarios:
- User saves from VSCode
- User saves from custom editor by clicking on the button in the UI
- User makes the change in custom editor and this change is auto-synced with local file
We want to catch a usage for each of these scenarios, I'm not sure if we can achieve that with separate metrics.

The way you are suggesting we would need to post 2 metrics for save type such as 
```
span.record({
  saveType: 'MANUAL_SAVE',
  source: 'VSCODE',
})
```
Can you confirm if we will be able to query a metric by multiple fields when constructing a dashboard? I.e. we'll want to detect detect save events originating from custom editor vs VSCode (and exclude `AUTO_SYNC` event)

Reply:
> The way you are suggesting we would need to post 2 metrics for save type such as
> 
> ```
> span.record({
>   saveType: 'MANUAL_SAVE',
>   source: 'VSCODE',
> })
> ```

By "2 metrics" do you mean "2 fields in 1 metric" (which aligns with the code example)? That's normal and idiomatic, and no problem for dashboards.